eclass/coreos-doc.eclass

```diff
@@ -0,0 +1,29 @@
+# Copyright 2015 CoreOS, Inc.
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+# @ECLASS: coreos-doc.eclass
+# @BLURB: utility function for including documentation
+
+# @FUNCTION: dodoc-coreos
+# @USAGE: [-r] <list of docs>
+dodoc-coreos() {
+	debug-print-function ${FUNCNAME} "$@"
+
+	local flags
+	if [[ "$1" == "-r" ]] ; then
+		flags="-r"
+		shift
+	fi
+
+	[[ "${#}" -lt 1 ]] && die "$0: at least one file needed"
+
+	insinto "/usr/share/coreos/doc/${P}/"
+	for f in "${@}" ; do
+		if [[ -s "${f}" ]] ; then
+			doins "${flags}" "${f}"
```

Comment:
without '-r' this will pass an empty string "" as the first argument. That seems not quite right but may be harmless. If you remove the quotes an empty string in ${flags} will not become an argument.